### PR TITLE
Fix failing Smoke Test Daily workflow

### LINKED ISSUE #31715: Daily smoke test workflow is failing

### PR DESCRIPTION
**Pull request by MrJnrman**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

**Background**: Our `smoke-test-daily.yml` file uses a matrix to run our E2E suite with various plugins installed. To achieve our goal, we had updated `GITHUB_REPOSITORY` ENV for each configuration defined by the matrix.

**Observation**: After some trial and error, I discovered that `GITHUB_REPOSITORY` is not updateable(or at least not obviously so and I found nothing in the docs).

**Solution**: This PR addresses this issue by setting a `PLUGIN_REPOSITORY` variable instead and updating the `upload-plugin.js` smoke test to be able to use it when set.

Closes #31715.

### How to test the changes in this Pull Request:

The steps below were already done against this PR and the results can be found here: https://github.com/woocommerce/woocommerce/runs/4958517809?check_suite_focus=true

1. Add a `.yml` file with the contents below. Its a stripped out version of the `smoke-test-daily.yml` workflow. Testing against `smoke-test-daily.yml` isn't viable since that workflow is executed on a schedule.
```
name: Smoke test daily Test
on:
  pull_request:
    branches:
      - trunk
    types:
      - labeled
jobs:
  build:
    name: Build zip for PR
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      - name: Build
        id: build
        uses: woocommerce/action-build@trunk
        env:
          BUILD_ENV: e2e

      - name: Upload PR zip
        uses: actions/upload-artifact@v2
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          name: woocommerce
          path: ${{ steps.build.outputs.zip_path }}
          retention-days: 7

  test-plugins:
    name: Smoke tests with ${{ matrix.plugin }} plugin installed
    runs-on: ubuntu-18.04
    needs: [build]
    strategy:
      fail-fast: false
      matrix:
        include:
          - plugin: 'WooCommerce Payments'
            repo: 'automattic/woocommerce-payments'
    steps:
      - name: Create dirs.
        run: |
          mkdir -p code/woocommerce
          mkdir -p package/woocommerce
          mkdir -p tmp/woocommerce
          mkdir -p node_modules

      - name: Checkout code.
        uses: actions/checkout@v2
        with:
          path: package/woocommerce

      - name: Install PNPM and install dependencies
        working-directory: package/woocommerce
        run: |
          npm install -g pnpm
          pnpm install

      - name: Load docker images and start containers.
        working-directory: package/woocommerce/plugins/woocommerce
        run: pnpx wc-e2e docker:up

      - name: Move current directory to code. We will install zip file in this dir later.
        run: mv ./package/woocommerce/plugins/woocommerce/* ./code/woocommerce

      - name: Download WooCommerce ZIP.
        uses: actions/download-artifact@v2
        with:
          name: woocommerce
          path: tmp

      - name: Extract and replace WooCommerce zip.
        working-directory: tmp
        run: |
          unzip woocommerce.zip -d woocommerce
          mv woocommerce/woocommerce/* ../package/woocommerce/plugins/woocommerce/

      - name: Install dependencies again
        working-directory: package/woocommerce
        run: |
          npm install -g pnpm
          pnpm install

      - name: Run tests command.
        working-directory: package/woocommerce/plugins/woocommerce
        env:
          WC_E2E_SCREENSHOTS: 1
          E2E_SLACK_TOKEN: ${{ secrets.SMOKE_TEST_SLACK_TOKEN }}
          E2E_SLACK_CHANNEL: ${{ secrets.RELEASE_TEST_SLACK_CHANNEL }}
          PLUGIN_REPOSITORY: ${{ matrix.private && secrets[matrix.repo] || matrix.repo }}
          PLUGIN_NAME: ${{ matrix.plugin }}
          GITHUB_TOKEN: ${{ secrets.E2E_GH_TOKEN }}
        run: |
          pnpx wc-e2e test:e2e tests/e2e/specs/smoke-tests/upload-plugin.js
          pnpx wc-e2e test:e2e
```

2. Add the `run: daily tests` label.
3. Observe that the `upload-plugin.js` test is described as `Upload and activate WooCommerce Payments from automattic/woocommerce-payments` to ensure that ENV variables are being picked up correctly.
4. Observe that the tests can successfully execute(failures might occur).

**Note**: If I don't get to it first, please remove the `run: daily tests` label using this [link](https://github.com/woocommerce/woocommerce/labels?q=run) if this PR is accepted. It was only created to test the `.yml` script above.

### CODE REVIEW

File: plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js [hunk 1]
```diff
@@ -13,30 +13,36 @@ const {
...
 
 let zipUrl;
 let pluginPath;
 
-utils.describeIf( GITHUB_REPOSITORY )( 'Upload and activate plugin', () => {
-	beforeAll( async () => {
-		zipUrl = await getLatestReleaseZipUrl( GITHUB_REPOSITORY, GITHUB_TOKEN );
-
-		pluginPath = await getRemotePluginZip( zipUrl, GITHUB_TOKEN );
+utils.describeIf( GITHUB_REPOSITORY || PLUGIN_REPOSITORY )(
```
Comment (rrennick) [inline plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js]:
```suggestion
utils.describeIf( repository )(
```

**Review by rrennick — CHANGES_REQUESTED** [review]
@MrJnrman This looks great except the one small suggestion I've made.

File: plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js (hunk 1 above)
Comment (MrJnrman) [reply plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js]: Thanks for the suggestion. I've made the change

**Review by rrennick — APPROVED** [review]
@MrJnrman Nice work on this.

**Pull request merged**